Cargo.toml

```diff
@@ -70,8 +71,24 @@ docs = ["approx", "serde", "rayon"]
 std = ["num-traits/std", "matrixmultiply/std"]
 rayon = ["dep:rayon", "std"]
 
+critical-section = ["portable-atomic/critical-section"]
```

Comment:
is critical section the right feature name for us? Maybe not

Reply:
What would recommend the name to be? Something like `no-std-enable-critical-section`?

[Heepless](https://github.com/rust-embedded/heapless), a very popular embedded rust project names the feature `portable-atomic-critical-section`
```toml
# Enable polyfilling of atomics via portable-atomic, using critical section for locking
portable-atomic-critical-section = ["dep:portable-atomic", "portable-atomic", "portable-atomic?/critical-section"]
```

Reply:
portable-atomic-critical-section sounds good then